packages/docmanager/src/widgetmanager.ts

```diff
@@ -184,7 +197,8 @@ export class DocumentWidgetManager implements IDisposable {
     if (!factory) {
       return undefined;
     }
-    let newWidget = this.createWidget(factory, context);
+    let newWidget = factory.clone(widget as IDocumentWidget, context);
```

Comment:
Let's just make the `cloneWidget` take the same thing that `createWidget` originally returned (i.e., `widget: IDocumentWidget` in the function arguments). Those are the widgets that have our private property anyway.

Reply:
I went ahead and moved that typing up to the function arguments for you.